internal/clients/team3/president.go

```diff
@@ -127,25 +150,33 @@ func (p *president) EvaluateAllocationRequests(resourceRequest map[shared.Client
 }
 
 func (p *president) SetTaxationAmount(islandsResources map[shared.ClientID]shared.ResourcesReport) shared.PresidentReturnContent {
+
+	if p.c.params.adv != nil {
+		val, done := p.c.params.adv.SetTaxationAmount(islandsResources)
+		if done {
+			return val
+		}
+	}
+
 	//decide if we want to run SetTaxationAmount
 	p.c.declaredResources = make(map[shared.ClientID]shared.Resources)
 	for island, report := range islandsResources {
 		if report.Reported {
 			p.c.declaredResources[island] = report.ReportedAmount
 		} else {
-			//TODO: read from params.config file once its available now i just hope they die of satanic attacc
-			p.c.declaredResources[island] = shared.Resources(666)
+			p.c.declaredResources[island] = shared.Resources(p.c.ServerReadHandle.GetGameConfig().IIGOClientConfig.AssumedResourcesNoReport)
 		}
 	}
 	gameState := p.c.BaseClient.ServerReadHandle.GetGameState()
-	resourcesRequired := 100.0 - float64(gameState.CommonPool)
-	if len(p.c.disasterPredictions) > int(p.c.ServerReadHandle.GetGameState().Turn) {
-		if disaster, ok := p.c.disasterPredictions[int(p.c.ServerReadHandle.GetGameState().Turn)][p.c.GetID()]; ok {
-			if disaster.TimeLeft != 0 {
-				resourcesRequired = disaster.Magnitude - float64(gameState.CommonPool)/float64(disaster.TimeLeft)
-			}
-		}
+	// Aim to have 100 in common pool after iigo run
+	resourcesRequired := math.Max((float64(p.c.getIIGOCost())+100.0)-float64(gameState.CommonPool), 0)
+	p.c.clientPrint("Resources required in common pool %f", resourcesRequired)
+
+	if len(p.c.globalDisasterPredictions) > int(p.c.ServerReadHandle.GetGameState().Turn) {
+		disaster := p.c.globalDisasterPredictions[int(p.c.ServerReadHandle.GetGameState().Turn)]
+		resourcesRequired += disaster.Magnitude - float64(gameState.CommonPool)/float64(disaster.TimeLeft+1)
```

Comment:
can disaster.TimeLeft ever be -1

Reply:
No idea really, I've added the checks now